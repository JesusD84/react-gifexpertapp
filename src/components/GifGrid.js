import React, { useEffect, useState } from 'react'

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=love&limit=10&api_key=Bmmv4eJ3ucVtu2v9ks8MlQdV79HgIUbn';
        const resp = await fetch(url);
        const { data } = await resp.json();
        
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }

        });

        setImages(gifs);

    }

    return (
        <div>
            <h3>{category}</h3>
            <ol>
                {
                    images.map(img => {
                        return <li key={img.id}>{img.title}</li>
                    })
                }
            </ol>
        </div>
    )
}
