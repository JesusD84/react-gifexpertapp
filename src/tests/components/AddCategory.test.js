import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"


describe('Pruebas en <AddCategory />', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de cambiar la caja de texto', () => {
        
        let input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        input.simulate('change', { target: { value } });
        
        input = wrapper.find('input');
        expect(input.prop('value')).toBe(value);

    });
    
    test('NO debe de postear la informacion con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();

    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const value = 'Hola Mundo';
        
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        const input = wrapper.find('input');

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect(input.prop('value')).toBe('');

    });
    
    

})
