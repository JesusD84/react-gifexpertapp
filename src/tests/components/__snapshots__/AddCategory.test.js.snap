// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Escribe los Gifs a buscar:
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
